src/urllib3/util/connection.py

```diff
@@ -87,7 +92,12 @@ def create_connection(
             # Break explicitly a reference cycle
             err = None
     else:
-        raise OSError("getaddrinfo returns an empty list")
+        message = "getaddrinfo returns an empty list"
+        if blocklist_applied:
+            blocklist_str = ", ".join(ip_addr_blocklist) if ip_addr_blocklist else ""
```

Comment:
Why do we need a ternary here? If we need one, something is seriously broken with your logic above